src/drivers/uavcan/sensors/battery.cpp

```diff
@@ -95,7 +95,7 @@ UavcanBatteryBridge::battery_sub_cb(const uavcan::ReceivedDataStructure<uavcan::
 	// battery.average_time_to_empty = msg.;
 	battery.serial_number = msg.model_instance_id;
 	battery.connected = true;
-	battery.system_source = msg.status_flags & uavcan::equipment::power::BatteryInfo::STATUS_FLAG_IN_USE;
+	battery.source = battery_status_s::BATTERY_SOURCE_POWER_MODULE;
```

Comment:
This is not true since a UAVCAN battery is not the power module. In the current implementation, this tells you if the battery is charging or discharging.

Reply:
Aha! What should I change it to?

Reply:
`battery.source = msg.status_flags & uavcan::equipment::power::BatteryInfo::STATUS_FLAG_IN_USE;`

Reply:
So I've removed the field `system_source` because it was not used anywhere from what I could tell and the description was not clear to me. This is now just a `source` field indicating what it is.

Reply:
I can only see the system_source being useful if a vehicle batteries that can be charged while attached.

Reply:
I would suggest to add it again if we have the implementation using it and the whole picture around it.